docs/src/developers_guide/contributing_tests.rst

```diff
@@ -0,0 +1,301 @@
+.. include:: ../common_links.inc
+
+.. _contributing_tests:
+
+***********************
+Contributing Iris Tests
+***********************
+
+.. _developer_pytest_categories:
+
+Test Categories
+===============
+
+There are two main categories of tests within Iris:
+
+- :ref:`testing.unit_test`
+- :ref:`testing.integration`
+
+Ideally, all code changes should be accompanied by one or more unit
+tests, and by zero or more integration tests.
+
+But if in any doubt about what tests to add or how to write them please
+feel free to submit a pull-request in any state and ask for assistance.
+
+
+.. _pytesting.unit_test:
+
+Unit Tests
+----------
+
+Code changes should be accompanied by enough unit tests to give a
+high degree of confidence that the change works as expected. In
+addition, the unit tests can help describe the intent behind a change.
+
+The docstring for each test module must state the unit under test.
+For example:
+
+    :literal:`"""Unit tests for the \`iris.experimental.raster.export_geotiff\` function."""`
+
+All unit tests must be placed and named according to the following
+structure:
+
+
+.. _pytesting.classes:
+
+Classes
```

Comment:
The examples in this section still show classes inheriting from `tests.IrisTest`, which is now deprecated.